README.md

```diff
@@ -1,2 +1,122 @@
+Historically, we have published major releases every 6-9 months. With our project now
+split across many subprojects, we expect to release subprojects separately and more
+often.
+
+This roadmap only includes mainline subproject under the Jupyter
+(https://github.com/jupyter) and IPython (https://github.com/ipython) GitHub orgs. We
+also have a number of newer, experimental subprojects in the Jupyter Incubator
+(https://github.com/jupyter-incubator) organization.
+
+## Notebook
+
+### Next (4.1)
+
+* Multi-cell selection and actions.
+* Notebook wide find and replace.
+* Command palette.
+* Restart kernel and run all.
+
+### Next+1 (5.0)
+
+* Prototype of the *Jupyter Workbench*:
+  - Initially, a new page in the existing notebook web-app that allows multiple
+    components on the same browser tab.
+  - Flexible, custimizable tabbed and paneled layout system.
+  - File browser.
+  - Text editor.
+  - Terminal.
+* All new frontend work is being done as small, loosely coupled npm packages.
+* The entire Jupyter Workbench will be a set of plugins (again just npm packages)
+  that exchange runtime APIs.
+* Application infrastucture (keyboard shortcuts, commands, menus, layout), view
+  layer, and other utilities (signals, properties, messages) provided by phosphorjs
+  (http://phosphorjs.github.io/).
+* Models and views that can be used on any web page in the following repos:
+  - https://github.com/jupyter/jupyter-js-services
+  - https://github.com/jupyter/jupyter-js-terminal
+  - https://github.com/jupyter/jupyter-js-editor
+  - https://github.com/jupyter/jupyter-js-filebrowser
+  - https://github.com/jupyter/jupyter-js-input-area
+  - https://github.com/jupyter/jupyter-js-output-area
+  - https://github.com/jupyter/jupyter-js-cells
+  - https://github.com/jupyter/jupyter-js-notebook
+* Wrapping of those models and views as plugins for the Jupyter Workbench
+  - https://github.com/jupyter/jupyter-js-services-plugin
+  - https://github.com/jupyter/jupyter-js-metaservice-plugin
+  - https://github.com/jupyter/jupyter-js-output-area-plugin
+  - https://github.com/jupyter/jupyter-js-notebook-plugin
+  - https://github.com/jupyter/jupyter-js-terminal-plugin
+  - https://github.com/jupyter/jupyter-js-editor-plugin
+  - https://github.com/jupyter/jupyter-js-filebrowser-plugin
+* Movement towards having stable, documented JavaScript APIs.
+* Ability to hide/show code and entire cells based on query syntax.
+
+### Next+2
+
+* New plugins for the Jupyter Workbench:
+  - Standalone output area.
+  - Ability to hook kernels up to text editors and output areas.
+  - Themes.
+
+### Future
+
+* New plugins for the Jupyter Workbench:
+  - Notebook.
+  - Variable inspector.
+  - UI for managing plugins.
+* Dashboarding of output areas and interactive widgets.
+* Replace existing pages in the notebook web-application by the Jupyter Workbench.
+* Interactive debugger.
+* Real-time collaboration on notebooks, text-editors and other plugins.
```

Comment:
Beyond about `N+1` major releases, I think it gets really difficult to predict which feature will get into which release. This is why I have used `Future` to indicate "other major releases in the future." I will add a note about that at the top of the document.

RT collab will take us a long time to finish fully - our thinking right now is that 5.0 of the notebook would be out shortly after 4.1 (very early 2016). Some parts of the UI may start to get RT collab soon though - but those would be separate npm packages that implement that - IOW plugins.